lib/elixir/src/elixir_expand.erl

```diff
@@ -1008,6 +1023,20 @@ format_error({op_ambiguity, Name, Arg}) ->
     "\"~ts ~ts\" looks like a function call but there is a variable named \"~ts\", "
     "please use explicit parentheses or even spaces",
   io_lib:format(Message, [Name, 'Elixir.Macro':to_string(Arg), Name]);
+format_error({invalid_clauses, Name}) ->
+  SupportMessage =
+    case Name of
+      'case' ->
+        "If you are trying to use the \"case\" structure, it should be similar to:\n"
+        "    case value do"
+        "      pattern -> :ok"
+        "    end";
+
+      _ ->
+        "Clauses are only valid for case, cond, fn, receive, else, catch or rescue."
```

Comment:
Clauses they are actually valid in any macro. For now I think we should drop the support message. :)